rancher/terraform-provider-rancher2
# Change tests

## Issue:  rancher/terraform-provider-rancher2#1148 

 
## Problem
 Tests were printing point address instead of json value.  
 
## Solution

Change the test functions that used  `!reflect.DeepEqual`to  use testify `assert.Equal` 


Example of testify output: 

```

=== RUN   TestFlattenClusterDriver
    structure_cluster_driver_test.go:61: 
        	Error Trace:	/home/user/go/src/terraform-provider-rancher2/rancher2/structure_cluster_driver_test.go:61
        	Error:      	Not equal: 
        	            	expected: map[string]interface {}{"active":true, "actual_url":"actual_urll", "builtin":true, "checksum":"XXXXXXXX", "name":"name", "ui_url":"ui_url", "url":"url", "whitelist_domains":[]interface {}{"domain1", "domain2"}}
        	            	actual  : map[string]interface {}{"active":true, "actual_url":"actual_url", "builtin":true, "checksum":"XXXXXXXX", "name":"name", "ui_url":"ui_url", "url":"url", "whitelist_domains":[]interface {}{"domain1", "domain2"}}
        	            	
        	            	Diff:
        	            	--- Expected
        	            	+++ Actual
        	            	@@ -2,3 +2,3 @@
        	            	  (string) (len=6) "active": (bool) true,
        	            	- (string) (len=10) "actual_url": (string) (len=11) "actual_urll",
        	            	+ (string) (len=10) "actual_url": (string) (len=10) "actual_url",
        	            	  (string) (len=7) "builtin": (bool) true,
        	Test:       	TestFlattenClusterDriver
        	Messages:   	Unexpected output from flattener.
--- FAIL: TestFlattenClusterDriver (0.00s)


Expected :map[string]interface {}{"active":true, "actual_url":"actual_urll", "builtin":true, "checksum":"XXXXXXXX", "name":"name", "ui_url":"ui_url", "url":"url", "whitelist_domains":[]interface {}{"domain1", "domain2"}}
Actual   :map[string]interface {}{"active":true, "actual_url":"actual_url", "builtin":true, "checksum":"XXXXXXXX", "name":"name", "ui_url":"ui_url", "url":"url", "whitelist_domains":[]interface {}{"domain1", "domain2"}}
<Click to see difference>


```

## Testing

 
## Engineering Testing
### Manual Testing


 Run it locally to ensure that it didn't change the tests behavior.
 
 Force an error and test the output. Then change it back to normal. 

### Automated Testing


## QA Testing Considerations

 
### Regressions Considerations
